fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBReturnedRecordLimitQueryTest.java

```diff
@@ -88,13 +90,13 @@ public void testComplexLimits2() throws Exception {
                     FDBQueriedRecord<Message> rec = cursor.next();
                     TestRecords1Proto.MySimpleRecord.Builder myrec = TestRecords1Proto.MySimpleRecord.newBuilder();
                     myrec.mergeFrom(rec.getRecord());
-                    assertEquals("odd", myrec.getStrValueIndexed());
                     assertEquals(0, myrec.getNumValue2());
+                    assertEquals("odd", myrec.getStrValueIndexed());
                     i += 1;
                 }
             }
             assertEquals(10, i);
-            assertDiscardedAtMost(18, context);
+            assertDiscardedAtMost(34, context);
```

Comment:
While the plan is better it seems that we discard more records. I am assuming that is because of prefetching but it would be nice to understand exactly why that is.

Reply:
Prefetching is not particularly relevant.

With the new plan, the entire index scan must be reversed to accomplish reverse sorting by the first field, which means that we do the odd-2's, then the odd-1's, then odd-0's. So we have to skip over those first two before getting to any matches and then we get 10 in a row. With the old single-field index plan, we go in reverse primary key order and every third matches, so we get the 10 we need faster.